pkgs/build-support/fetchpypilegacy/default.nix

```diff
@@ -18,28 +33,31 @@
   name ? null,
 }:
 let
-  urls' = urls ++ lib.optional (url != null) url;
+  urls' = urls ++ optional (url != null) url;
 
-  pathParts = lib.filter ({ prefix, path }: "NETRC" == prefix) builtins.nixPath;
-  netrc_file =
-    if (pathParts != [ ])
-    then (lib.head pathParts).path
-    else "";
+  pathParts = filter ({ prefix, path }: "NETRC" == prefix) builtins.nixPath;
+  netrc_file = if (pathParts != [ ]) then (head pathParts).path else "";
 
 in
 # Assert that we have at least one URL
-assert urls' != [ ]; runCommand file
-  ({
-    nativeBuildInputs = [ python3 ];
-    impureEnvVars = lib.fetchers.proxyImpureEnvVars;
-    outputHashMode = "flat";
-    # if hash is empty select a default algo to let nix propose the actual hash.
-    outputHashAlgo = if hash == "" then "sha256" else null;
-    outputHash = hash;
-    NETRC = netrc_file;
-  }
-  // (lib.optionalAttrs (name != null) {inherit name;}))
+assert urls' != [ ];
```

Comment:
Future: use lib.assertMsg to integrate the comment into the guidance.

Reply:
Indeed. Not touching this now as this diff is a formatting change.